.github: add release template for RCs from master branch (#11)
cilium/release: .github/ISSUE_TEMPLATE/release_template_rc_master.md

---
name: Release a new RC version of Cilium from master branch
about: Create a checklist for an upcoming release
title: 'vX.Y.Z-rcW release'
labels: kind/release
assignees: ''

---

## Pre-release

- [ ] Announce in Cilium slack channel #launchpad: `Starting vX.Y.Z-rcW release process :ship:`
- [ ] Create a thread for that message and ping current top-hat to not merge any
      PRs until the release process is complete.
- [ ] Change directory to the local copy of Cilium repository.
- [ ] Make sure docker is running.
- [ ] Export a `GITHUB_TOKEN` that has access to the repository.
- [ ] Check that there are no [release blockers] for the targeted release version
- [ ] Consider building new [cilium-runtime images] and bumping the base image
      versions on this branch:
  - Cilium v1.10 or later:
     Modify the `FORCE_BUILD` environment value in the `images/runtime/Dockerfile` to force a rebuild.
     [Instructions](https://docs.cilium.io/en/latest/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images)
  - Cilium v1.7 to v1.9:

>      [Re-trigger a build in quay.io](https://docs.cilium.io/en/v1.9/contributing/development/images/#update-cilium-builder-and-cilium-runtime-images)

Author: Yes, this can be removed because this is RC from master branch.

  - [ ] Check the output of the [chart workflow] and see if the test was
        successful.
- [ ] Check [read the docs] configuration:
    - [ ] Set a new build as active and hidden in [active versions].
    - [ ] Deactivate previous RCs.
    - [ ] Update algolia configuration search in [docsearch-scraper-webhook].
- [ ] Check draft release from [releases] page
  - [ ] Update the text at the top with 2-3 highlights of the release
  - [ ] Mark the checkbox of "This is a pre-release"
  - [ ] Publish the release
- [ ] Announce the release in #general on Slack.
Text template for the first RC:
```
*Announcement* :tada: :tada:

:cilium-new: *Cilium release candidate vX.Y.Z-rcW has been released:*
https://github.com/cilium/cilium/releases/tag/vX.Y.Z-rcW

This kicks off the release train that leads us towards vX.Y final version in the coming weeks. There are [vX.Y.Z-rcW OSS docs](https://docs.cilium.io/en/vX.Y.Z-rcW) available if you want to pull this version & try it out.
```
Text template for the next RCs:
```
*Announcement* :tada: :tada:

:cilium-new: *Cilium release candidate vX.Y.Z-rcW has been released:*
https://github.com/cilium/cilium/releases/tag/vX.Y.Z-rcW

> Thank you for the bugs reported found in the previous RC. There are [vX.Y.Z-rcW OSS docs](https://docs.cilium.io/en/vX.Y.Z-rcW) available if you want to pull this version & try it out.

Reviewer: nit, RCs are partly about bugs, but also about docs improvements, other feature freeze exception PRs, etc. Broadly, contributions.

```suggestion
Thank you for testing and contributing to the previous RC. There are [vX.Y.Z-rcW OSS docs](https://docs.cilium.io/en/vX.Y.Z-rcW) available if you want to pull this version & try it out.
```

```

> ## Post-release

Reviewer: Are there any post-release steps?

Author: No, somehow I forgot to remove this.
